### ISSUE TITLE
(Rogue) Vanish

### ISSUE THREAD

**Issue by Bzmode** [ccshiro/cc-buglist#636]
http://www.twitch.tv/faleene/c/6797980
Vanish doesn't drop enemy target, allowing me to cheap shot after target vanishes. Also breaks player frame permanently.

**Comment by ghost** [comment]
Vanish didn't drop enemy target on retail either, for a very very short period of time, allowing you to cast spells through vanish\* if you timed it right. I definitely used to do this on retail during WotLK with sap/cheap shot and vanish/shadow dance, and blind. I was aware of this existing back in TBC too, maybe I'm wrong. That being said, I'm not sure if it should be this easy (or if this is the correct behavior at all). I'll look for some videos. Target frame breaking is definitely a bug, though, as I don't recall this happening on retail.

@ceyscro targeting enemy != having a target frame, there used to be a short period during which you'd have a target but not have a target frame after the target would vanish. I suppose the target frame was getting dropped before the actual target would change to 'none'.

EDIT: https://www.youtube.com/watch?v=aheVMgpjpkQ This is akrios' dueling video from WotLK

EDIT2: Meant to say Vanish but said Blind instead. Added more info.

**Comment by ceyscro** [comment]
Yeah as SorcerersApprentice said, for a short period of time you see target while vanishing.
Absolutely normal behavior, this is what allows vanish to "absorb" stuns or to stuns to break vanish.
Here a screen from retail video : 
http://image.noelshack.com/fichiers/2015/17/1429818246-vanish.png

**Comment by Peonstomperx** [comment]
Cheap Shotting vanishes is working as intended.
The target bug(invisible frame) is pretty well known on AT amongst rogues and usually happens when you spam step+cheap shot on arena1/2/3 as they open on something.

I would assume the cause has to do with targetting a unit that's currently invisible and stealth delay, maybe it's a client side bug.

**Comment by ghost** [comment]
@Peonstomperx I wonder if a macro such as the one below would fix that bug:
/cleartarget
/targetlasttarget

**Comment by ccshiro** [comment]
The frame does bug if you cheap shot a rogue as he vanishes. No other spells seems to bug it (tried kidney shot, sap, sinister strike). Not sure exactly why it bugs, but it's 100% reproducible.

**Comment by lawzftw** [comment]
About the frame bug, I remember asking about it on AT (wotlk though), and they said it's client side. So this bug probably also existed on retail. I only had it on focus frames though, so it might not be the same bug. For every way to "disappear" (vanish, stealth, meld etc), we kept the focus, and when they were visible we still had the focus, but no frame (as here).

**Comment by Nain1** [comment]
Should Vanish not provide immunity to stuns?  At least when used?

**Comment by ceyscro** [comment]
Depending of timing. Vanish can either avoid a stun or be broken by the stun (rogue take the stun).

**Comment by darkcranio** [comment]
I remember charging into vanishes, was fun.

**Comment by ccshiro** [comment]
I can't figure out why the bug happens. Will move it to low priority since it's pretty time consuming and not that big of an impact (the trigger window is pretty small and reselecting the target gets rid of the bug).
